### PR TITLE
Pin sidebar with a button

### PR DESCRIPTION
**Pull request by adaszyn**
## Hey, I just made a Pull Request!

Pin sidebar with a button. Keep the previous state in `localStorage`.

pin-sidebar

**Dark mode**:
image

**Light mode**:
image

#### :heavy_check_mark: Checklist

- [X] All tests are passing `yarn test`
- [X] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [X] Prettier run on changed files
- [X] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by adaszyn** [comment]
Fixes https://github.com/spotify/backstage/issues/907

**Review by Rugvip — COMMENTED** [review]
👍, but could you take a stab at moving this into the sidebar component itself? Or some other way of getting it into @backstage/core. We don't want to have any code that is supposed to be reused in the app package.

**Comment by stefanalund** [comment]
This works exactly as I imagined it, great work @adaszyn

**Comment by stefanalund** [comment]
@Rugvip wanna take another look?

File: packages/theme/src/themes.ts [hunk 1]
```diff
@@ -56,6 +56,10 @@ export const lightTheme = createTheme({
     link: '#0A6EBE',
     gold: yellow.A700,
     sidebar: '#171717',
+    pinSidebarButton: {
```
Comment (Rugvip) [inline packages/theme/src/themes.ts:59]: this is a bit too specific imo. Maybe we can try iconForeground + iconBackground, or `{ icon: { foreground, background }}`? Could even just put there colors in the component and switch on palette.type too tbh.

**Review by Rugvip — APPROVED** [review]
👍 , the theme stuff isn't a solved area, so happy to merge. But maybe you have some ideas? :grin:

**Pull request merged**